update tikv-control.md (#3083)
pingcap/docs: tikv-control.md

# TiKV Control User Guide

TiKV Control (`tikv-ctl`) is a command line tool of TiKV, used to manage the cluster. Its installation directory is as follows:

> * If the cluster is deployed using TiDB Ansible, it exists in the `resources/bin` subdirectory under the `ansible` directory.

Reviewer: ```suggestion
* If the cluster is deployed using TiDB Ansible, `tikv-ctl` directory is in the `resources/bin` subdirectory under the `ansible` directory.
```

> * If the cluster is deployed using TiUP, it exists in `~/.tiup/components/ctl/{VERSION}/` directory.

Reviewer: ```suggestion
* If the cluster is deployed using TiUP, `tikv-ctl` directory is in the `~/.tiup/components/ctl/{VERSION}/` directory.
```

> [TiUP](https://github.com/pingcap-incubator/tiuptiup) is a deployment tool introduced later than `tidb-ansible`, and its usage is more simplified. `tikv-ctl` is also integrated in the `tiup` command. Execute the following command to call the `tikv-ctl` tool:

Reviewer: ```suggestion
[TiUP](https://github.com/pingcap-incubator/tiuptiup) is a deployment tool introduced later than TiDB Ansible, and its usage is simpler. `tikv-ctl` is also integrated in the `tiup` command. Execute the following command to call the `tikv-ctl` tool:
```

{{< copyable "shell-regular" >}}

```bash
tiup ctl tikv
```

```
Starting component `ctl`: ~/.tiup/components/ctl/v4.0.0-rc.2/ctl tikv
TiKV Control (tikv-ctl)
Release Version:   4.0.0-rc.2
Edition:           Community
Git Commit Hash:   2fdb2804bf8ffaab4b18c4996970e19906296497
Git Commit Branch: heads/refs/tags/v4.0.0-rc.2
UTC Build Time:    2020-05-15 11:58:49
Rust Version:      rustc 1.42.0-nightly (0de96d37f 2019-12-19)
Enable Features:   jemalloc portable sse protobuf-codec
Profile:           dist_release
```

> You can add the appropriate parameters and subcommands behind `tiup ctl tikv`.

Reviewer: ```suggestion
You can add corresponding parameters and subcommands after `tiup ctl tikv`.
```

The `tombstone` command is usually used in circumstances where the sync-log is not enabled, and some data written in the Raft state machine is lost caused by power down.

> In a TiKV instance, you can use this command to set the status of some Regions to Tombstone. Then when you restart the instance, those Regions are skipped to avoid the restart failure caused by damaged Raft state machines of those Regions. Those Regions need to have enough healthy replicas in other TiKV instances to be able to continue the reads and writes through the Raft mechanism.

Reviewer: ```suggestion
In a TiKV instance, you can use this command to set the status of some Regions to tombstone. Then when you restart the instance, those Regions are skipped to avoid the restart failure caused by damaged Raft state machines of those Regions. Those Regions need to have enough healthy replicas in other TiKV instances to be able to continue the reads and writes through the Raft mechanism.
```

> In a TiKV instance, you can use this command to set the status of some Regions to Tombstone. Then when you restart the instance, those Regions are skipped so as to avoid the failure to restart because those Regions's Raft state machine are damaged. Those Regions need to have enough healthy replicas in other TiKV instances to be able to continue writing and reading through the Raft mechanism.

Reviewer: ```suggestion
In a TiKV instance, you can use this command to set the status of some Regions to Tombstone. Then when you restart the instance, those Regions are skipped to avoid the restart failure caused by damaged Raft state machines of those Regions. Those Regions need to have enough healthy replicas in other TiKV instances to be able to continue the reads and writes through the Raft mechanism.
```

> Under normal circumstances, you can remove the corresponding Peer of this Region using `remove-peer` command:

Reviewer: ```suggestion
In general cases, you can remove the corresponding Peer of this Region using the `remove-peer` command:
```

{{< copyable "shell-regular" >}}

```shell
pd-ctl operator add remove-peer <region_id> <store_id>
```

> Then use the `tikv-ctl` tool to set a Region to tombstone on the corresponding TiKV instance to skip the health check for this Region at startup:

Reviewer: ```suggestion
Then use the `tikv-ctl` tool to set a Region to Tombstone on the corresponding TiKV instance to skip the health check for this Region at startup:
```

> Then use the tikv-ctl to set a Region to Tombstone in the corresponding TiKV instance so it will skip this Region's health check at startup:

Reviewer: ```suggestion
Then use the `tikv-ctl` tool to set a Region to tombstone on the corresponding TiKV instance to skip the health check for this Region at startup:
```

{{< copyable "shell-regular" >}}

```shell
tikv-ctl --db /path/to/tikv/db tombstone -p 127.0.0.1:2379 -r <region_id>
```

```
success!
```

> However, in some cases, when it is not convenient to remove this Peer of this Region from PD, you can specify the `--force` option of tikv-ctl to force it to tombstone:

Reviewer: ```suggestion
However, in some cases, you cannot easily remove this Peer of this Region from PD, so you can specify the `--force` option in `tikv-ctl` to forcibly set the Peer to tombstone:
```
